### ISSUE TITLE
accessibilityLabel of non-text children does not propagate to parent component when focused

### ISSUE THREAD

**Issue by JakovS** [microsoft/react-native-windows#8297]
## Environment
1. `npx react-native --version`: 
4.13.0
2. `npx react-native info`:
System:
    OS: Windows 10 10.0.19043
    CPU: (12) x64 Intel(R) Xeon(R) CPU E5-1650 v3 @ 3.50GHz
    Memory: 13.24 GB / 31.91 GB
  Binaries:
    Node: 12.9.1 - C:\Program Files\nodejs\node.EXE
    Yarn: Not Found
    npm: 6.10.2 - C:\Program Files\nodejs\npm.CMD
    Watchman: Not Found
  SDKs:
    Android SDK: Not Found
    Windows SDK:
      AllowDevelopmentWithoutDevLicense: Enabled
      AllowAllTrustedApps: Enabled
      Versions: 10.0.16299.0, 10.0.17134.0, 10.0.17749.0, 10.0.17763.0, 10.0.18362.0, 10.0.19041.0
  IDEs:
    Android Studio: Not Found
    Visual Studio: 15.9.28307.1500 (Visual Studio Enterprise 2017), 16.10.31402.337 (Visual Studio Enterprise 2019)
  Languages:
    Java: Not Found
    Python: 2.7.15 - C:\Users\user\.windows-build-tools\python27\python.EXE
  npmPackages:
    @react-native-community/cli: 4.13.0 => 4.13.0
    react: 16.13.1 => 16.13.1
    react-native: 0.63.3 => 0.63.3
    react-native-windows: 0.63.14 => 0.63.14
  npmGlobalPackages:
    *react-native*: Not Found

## Steps To Reproduce
Provide a detailed list of steps that reproduce the issue.

1. Create a focusable view without accessibility label
2. Populate the view with subviews with accessilityLabel and text components
3. Focus root component with narrator
4. Notice it reads text component contents, but not accessibility label of child component

## Expected Results
Reads the text and closest accessibility labels for child components on focus.

## Snack, code example, screenshot, or link to a repository:
Please provide a Snack (https://snack.expo.io/), a link to a repository on GitHub, or provide a minimal code example that reproduces the problem.
You may provide a screenshot of the application if you think it is relevant to your bug report.
Here are some tips for providing a minimal example: https://stackoverflow.com/help/mcve

Simplified example, assume that children are external components that produce this as output.
```
<View focusable={true}>
   <View accessibilityLabel={"Label One"}/>
   <Text>{"Label Two"}</Text>
   <View accessibilityLabel={"Label Three"}/>
</View>
```
Expected focusing outer view to read `Label One, Label Two, Label Three`.
Currently it reads: `Label Two`

Current workaround is to provide child components a callback function that constructs label of parent component or change the code where all info is gathered by parent component and then prop drilled to children.

Example usage would be a button that has a product title and a connected star rating icon component (which loads async). Focusing on this button would have Narrator read product title and star rating.

**Comment by asklar** [comment]
Discussed this offline, this seems like something the platform should not provide as it is scenario specific (if a control is not focusable, then it is up to its parent to decide when/how to read it)

**Comment by chrisglein** [comment]
I think we've resolved that this is by design. If there's any comments from the offline conversations that might help people looking at this issue in the future, put them here.
